newrelic/metric_test.go

```diff
@@ -0,0 +1,106 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package newrelic
+
+import (
+	"context"
+	"io/ioutil"
+	"math"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/google/ts-bridge/record"
+	"google.golang.org/appengine/aetest"
+)
+
+var testCtx context.Context
+
+func TestMain(m *testing.M) {
+	var done func()
+	var err error
+	testCtx, done, err = aetest.NewContext()
+	if err != nil {
+		panic(err)
+	}
+
+	code := m.Run()
+	done()
+	os.Exit(code)
+}
+
+// checks that `value` is close to `target` within `margin`.
+func timeWithin(value, target time.Time, margin time.Duration) bool {
+	return math.Abs(value.Sub(target).Seconds()) <= margin.Seconds()
+}
+
+func mustUnmarshalText(s string, pb proto.Message) {
+	if err := proto.UnmarshalText(s, pb); err != nil {
+		panic(err)
+	}
+}
+
```

Comment:
I don't think `timeWithin` or `mustUnmarshalText` are actually used. Or did you want to add more tests before committing this? :)

Reply:
I mean - there should always be more tests... do I *want* to?